imageregistry/v1/types_imagepruner.go

```diff
@@ -44,13 +42,13 @@ type ImagePrunerSpec struct {
 	// +optional
 	Suspend *bool `json:"suspend,omitempty" protobuf:"bytes,2,opt,name=suspend"`
 	// keepTagRevisions specifies the number of image revisions for a tag in an image stream that will be preserved.
-	// Defaults to 5.
+	// Defaults to 3.
 	// +optional
 	KeepTagRevisions *int `json:"keepTagRevisions,omitempty" protobuf:"bytes,3,opt,name=keepTagRevisions"`
 	// keepYoungerThan specifies the minimum age of an image and its referrers for it to be considered a candidate for pruning.
-	// Defaults to 96h (96 hours).
+	// Defaults to 60m (60 minutes).
 	// +optional
-	KeepYoungerThan *time.Duration `json:"keepYoungerThan,omitempty" protobuf:"varint,4,opt,name=keepYoungerThan,casttype=time.Duration"`
+	KeepYoungerThan *metav1.Duration `json:"keepYoungerThan,omitempty" protobuf:"bytes,4,opt,name=keepYoungerThan"`
```

Comment:
/cc @sttts @deads2k 

My gut is that we can't do this, as the data is currently serialized as JSON integers.
I think we need to deprecate `KeepYoungerThan` and make a new field - `KeepYoungerThanDuration`

Reply:
yes, I think that's right.

One could maybe wrap `metav1.Duration` into a local type with a custom unmarshaller that understands the old serialization.

Reply:
> yes, I think that's right.
> 
> One could maybe wrap `metav1.Duration` into a local type with a custom unmarshaller that understands the old serialization.

That only resolves half the problem.  Any other consumer won't understand the resulting serialization if we change it.

Reply:
> /cc @sttts @deads2k
> 
> My gut is that we can't do this, as the data is currently serialized as JSON integers.
> I think we need to deprecate `KeepYoungerThan` and make a new field - `KeepYoungerThanDuration`

Can you try to come up with a compatibility plan for this new field that you would add?  I think it needs to consider...

1. old consumer which removes the new field and modifies the old
2. old consumer which keeps the new field and modifies the old
3. new consumer which keeps the old field and modifies the new
4. case where new object is not written (so admission isn't involved) and new consumer attempts a read

Reply:
Is this set with a default value by users today?  Or would this be empty on a fresh install?

Reply:
It's empty on a fresh install.